src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -133,6 +133,11 @@ private void delete(Path p) throws IOException {
             }
         }
         try {
+            if (isWindows()) {
+                // Windows throws an access denied exception when deleting read-only files
+                boolean ok = p.toFile().setWritable(true);
```

Comment:
Fine though if this ever fails it will do so in an opaque fashion; using NIO APIs might be better.